library/src/main/kotlin/com/connectrpc/ProtocolClientInterface.kt

```diff
@@ -110,3 +110,11 @@ interface ProtocolClientInterface {
         methodSpec: MethodSpec<Input, Output>,
     ): ClientOnlyStreamInterface<Input, Output>
 }
+
+fun Headers.toLowercase(): Headers {
```

Comment:
Some places were using `mapKeys` to turn keys to lower-case. But that is subtly incorrect if two different keys map to the same key after lower-casing, like `content-type` and `Content-Type`. So now we correctly merge the values if there are conflicting mixed-case keys.

This formulation is so that we don't actually need to compute intermediate collections, and basically create and process multiple maps to compute the end result. But maybe an imperative loop would be more clear?